pkg/distributor/distributor.go

```diff
@@ -51,6 +52,8 @@ var (
 		Name:      "distributor_lines_received_total",
 		Help:      "The total number of lines received per tenant",
 	}, []string{"tenant"})
+
+	maxLabelCacheSize = 10000
```

Comment:
```suggestion
	maxLabelCacheSize = 100000
```

I think 100k might be a better starting point for this. Unlike ingesters, distributors handle _all_ incoming traffic and are therefore exposed to many more unique label sets.